plugins/postcss-design-tokens/test/basic.css

```diff
@@ -0,0 +1,54 @@
+@import url('./imported.css');
```

Comment:
I love this test!

Could we have this sort of test too? Is it even supported?

```pcss
/* styles.css */
@import url('./vars.css');
@import url('./base.css');
```

```pcss
/* vars.css */
@design-tokens url('./tokens/basic.json') format('style-dictionary3');

:root {
  --ff-primary: design-token('font.family.base');
}
```

```pcss
/* base.css */
.is-color-base {
  color: design-token('color.font.base');
}
```

I think this is the natural path on mid to large CSS codebases, you have a single file that sets everything up and everything else is kept separated. This can also alleviate if you need `styles-dark.css` as a different entrypoint that just uses a different `var.css` file.

Note that `base.css` does not need to import design-tokens anymore which I don't think is showcased in these tests (could be wrong since I'm reading top to bottom and doing a thorough read 😂

Reply:
> This can also alleviate if you need styles-dark.css as a different entrypoint that just uses a different var.css file.

This should not be needed and I think it would even best be avoided.
By using `when` other tools (editor) can also be aware of conditional tokens.

```css
@design-tokens url('./tokens/basic.json') format('style-dictionary3');
@design-tokens url('./tokens/basic--theme-a.json') format('style-dictionary3') when('theme-a');

.foo {
  color: design-token('something.from.basic');
}
```

An editor plugin should understand `when` and give you feedback that `color` can have one of two possible values. This will give you greater insights into the design system.

------

> I think this is the natural path on mid to large CSS codebases, you have a single file that sets everything up and everything else is kept separated.

> Note that base.css does not need to import design-tokens anymore which I don't think is showcased in these tests (could be wrong since I'm reading top to bottom and doing a thorough read

True, this just works when using `postcss-import`.
But it also places extra burden on editor plugins.

They might not be able to reliably determine how you are building your CSS and which imports apply.

```
- entrypoint
  - a (has @design-tokens rules)
  - b
    - c (active file)
```

Not every tool/editor can figure out that everything comes together in `entrypoint` when you have `c` open as `c` doesn't reference `entrypoint` at all.

Adding a test is not really needed as it is not a feature we provide, but part of `postcss-import`.

Reply:
>An editor plugin should understand when and give you feedback that color can have one of two possible values. This will give you greater insights into the design system.

Example implementation of this :

- hover on `design-token` shows a table with all possible values and their conditions
- you can click through on the values to go to the tokens file

<img width="823" alt="Screenshot 2022-05-22 at 18 12 30" src="https://user-images.githubusercontent.com/11521496/169705065-18565a38-c585-44e3-805c-617b3ef8276d.png">

Reply:
All of this makes sense!